### DISCUSSION TITLE
Image frame count

### DISCUSSION THREAD

**Discussion by danielebogo** [onevcat/Kingfisher#1645] [General]
@onevcat Is it possible to store the animated image frame count when an animated image is created?

When `func animatedImage(data: Data, options: ImageCreatingOptions) -> KFCrossPlatformImage?` is called, the `GIFAnimatedImage` has the `images` array populated, so it's easy to just store the frame count.

If the no `preloadAll` or `onlyFirstFrame` is used, then we can retrieve the same info from the `imageSource` with `CGImageSourceGetCount(imageSource)`.

SDWebImage has this value stored in the image and it's retrieved by each decoder.

I can push a PR if necessary.

**Comment by onevcat** [comment]
Thanks for it. #1647 has been merged and a new version (6.2) was released for this.

**Comment by iphonehamza** [comment]
Yes, it's possible to store the animated image frame count when an animated image is created. When animatedImage(data:options:) is called, the GIFAnimatedImage object will have its images array populated, so it's easy to store the frame count. If noPreloadAll or onlyFirstFrame options are used, you can retrieve the frame count using CGImageSourceGetCount(imageSource). This is similar to how SDWebImage handles it. A pull request (PR) would be appreciated for this change.

For more information on handling phone repairs and issues, check out this guide on how to [fix a phone that won't charge](https://www.doctormobile.ca/how-to-fix-a-phone-that-wont-charge).

**Comment by amjad789789** [comment]
Pets Best Insurance: The Ultimate Guide to Protecting Your Furry Friend

Pets Best Insurance is a leading provider of pet insurance in the United States, offering comprehensive coverage for dogs and cats. With various plans and flexible options, it ensures that pet owners can afford the best medical care for their furry companions without financial stress.
Why Pet Insurance is Important?

Pet insurance provides financial security by covering unexpected veterinary bills. Just like health insurance for humans, pet insurance helps reduce the burden of high medical costs in case of accidents, illnesses, or routine care. Here are some reasons why pet insurance is essential:

    Veterinary costs are rising every year.
    Emergency treatments can be expensive.
    Helps in managing chronic conditions like diabetes or arthritis.
    Peace of mind knowing your pet will get the best care.

How Does Pets Best Insurance Work?

Pets Best operates on a reimbursement model. Pet owners pay the vet bill upfront and then submit a claim to Pets Best for reimbursement. The percentage of reimbursement depends on the policy chosen. The process typically includes:

    Visiting any licensed veterinarian in the U.S.
    Paying the bill for the treatment.
    Submitting a claim online or via the mobile app.
    Receiving reimbursement based on the selected plan.

Coverage Options Offered by Pets Best

Pets Best offers a variety of plans tailored to different needs and budgets. Here are the main coverage options:
1. Accident-Only Plan

This plan covers emergency situations like injuries from accidents, broken bones, or poisoning. It is ideal for pet owners looking for a low-cost insurance option.
[https://petsmartsmile.com](url)

**Comment by arian677** [comment]
Nice to see #1647 merged successfully. This release could be valuable for projects focused on **[PDGF receptor](https://www.betalifesci.com/products/recombinant-human-interleukin-2-il2-protein-blt-01506p)** data processing and analysis.
